protocol/app/upgrades.go

```diff
@@ -5,20 +5,19 @@ import (
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	v0_2_2 "github.com/dydxprotocol/v4-chain/protocol/app/upgrades/v0.2.2"
 )
 
 // setupUpgradeHandlers registers the upgrade handlers to perform custom upgrade
 // logic and state migrations for software upgrades.
 func (app *App) setupUpgradeHandlers() {
-	for _, upgrade := range Upgrades {
-		if app.UpgradeKeeper.HasHandler(upgrade.UpgradeName) {
-			panic(fmt.Sprintf("Cannot register duplicate upgrade handler '%s'", upgrade.UpgradeName))
-		}
-		app.UpgradeKeeper.SetUpgradeHandler(
-			upgrade.UpgradeName,
-			upgrade.CreateUpgradeHandler(app.ModuleManager, app.configurator),
```

Comment:
Probably need to update `CreateUpgradeHandler` so that it's more flexible to support different keepers. will think about that more

Reply:
Is passing in a reference to app itself a reasonable option?

Reply:
yeah I think that's a good idea.